libgadget/sfr_eff.c

```diff
@@ -324,9 +323,9 @@ cooling_direct(int i) {
     }
 #endif
 
-    struct UVBG uvbg;
-    GetParticleUVBG(i, &uvbg);
-    unew = DoCooling(unew, SPHP(i).Density * All.cf.a3inv, dtime, &uvbg, &ne, SPHP(i).Metallicity);
+    double redshift = 1./All.Time - 1;
+    struct UVBG uvbg = get_particle_UVBG(redshift, P[i].Pos);
```

Comment:
The function is no longer using a particle, so rename it to get_UVBG?

Reply:
get_local_UVBG?